### PR TITLE
Adapt the github workflow to run on nektos/act

### PR DESCRIPTION
**Pull request by devreal**
See https://github.com/nektos/act

Not sure we want to merge it. I'm still seeing the following errors for the unittests:

```
| 25: terminate called after throwing an instance of 'std::runtime_error'
| 25:   what():  locale::facet::_S_create_c_locale name not valid
```

That only occurs on the core-unittests though, all other tests succeed on my machine.

Signed-off-by: Joseph Schuchart <schuchart@icl.utk.edu>

### CODE REVIEW

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -41,7 +44,16 @@ jobs:
     - name: Create Build Environment
       # Some projects don't allow in-source building, so create a separate build directory
       # We'll use this as our working directory for all subsequent commands
-      run: cmake -E make_directory ${{github.workspace}}/build
+      run: |
+          echo "tzdata tzdata/Areas select America" > ubuntu-preseed.txt
```
Comment (evaleev) [inline .github/workflows/cmake.yml:48]: this should go into ubuntu package install section, messes up the macOS builds

File: .github/workflows/cmake.yml [hunk 2]
```diff
@@ -41,7 +44,16 @@ jobs:
...
+          echo "tzdata tzdata/Zones/Etc select UTC" >> ubuntu-preseed.txt
+          echo "tzdata tzdata/Zones/America select New_York" >> ubuntu-preseed.txt
+          debconf-set-selections ubuntu-preseed.txt
```
Comment (evaleev) [inline .github/workflows/cmake.yml:51]: sounds like this also needs `sudo`?

**Review by evaleev — CHANGES_REQUESTED** [review]

File: .github/workflows/cmake.yml (hunk 2 above)
Comment (devreal) [reply .github/workflows/cmake.yml]: Removed, didn't work and should instead be fixed by removing the `setlocale` call in the unit test.

File: .github/workflows/cmake.yml (hunk 1 above)
Comment (devreal) [reply .github/workflows/cmake.yml]: I had to move the make directory creation after the package installation to make sure we have CMake available.

**Review by evaleev — APPROVED** [review]

**Pull request merged**